src/middlewared/middlewared/plugins/security/update.py

```diff
@@ -0,0 +1,70 @@
+import re
+
+import middlewared.sqlalchemy as sa
+
+from middlewared.schema import accepts, Bool, Dict, Int, Patch
+from middlewared.service import CallError, ConfigService, private
+from middlewared.utils import run
+
+
+RE_FIPS = re.compile(r'name:\s+OpenSSL FIPS Provider')
+
+
+class SystemSecurityModel(sa.Model):
+    __tablename__ = 'system_security'
+
+    id = sa.Column(sa.Integer(), primary_key=True)
+    enable_fips = sa.Column(sa.Boolean(), default=False)
+
+
+class SystemSecurityService(ConfigService):
+
+    class Config:
+        cli_namespace = 'system.security'
+        datastore = 'system.security'
+        namespace = 'system.security'
+
+    ENTRY = Dict(
+        'system_security_entry',
+        Bool('enable_fips', required=True),
```

Comment:
Do we need to add "reboot" Boolean and "reboot_timeout" Int schema objects to automatically reboot system if they enable it? If we decide to do this, we need to handle HA systems.

Reply:
I don't think we should reboot ourselves, we don't do the same when grub configuration is changed 🤔 
Probably we can make that clear in the UI